Aspire BudgetingTests/GoogleSheetsValidatorTests.swift

```diff
@@ -0,0 +1,156 @@
+
+    let validator = GoogleSheetsValidator(sheetsService: mockService, sheetsQuery: query)
+    validator.validate(file: mockFile, for: mockUser)
+
+    let exp = XCTestExpectation()
+    exp.expectedFulfillmentCount = 2
+
+    validatorSink = validator.currentState.sink {
+      switch $0 {
+      case .isLoading:
+        exp.fulfill()
+
+      case .error(let error):
+        XCTAssertEqual(error as NSError, mockError)
+        exp.fulfill()
+      default:
+        XCTFail()
+      }
+    }
+
+    wait(for: [exp], timeout: 1)
+  }
+
+  func testValidationSuccess() {
+    let sheet1 = MockSheet(properties:
+                            createSheetProperties(id: 50, title: "Sheet1"))
+    let sheet2 = MockSheet(properties:
+                            createSheetProperties(id: 42, title: "Sheet2"))
+
+    let nr1 = GTLRSheets_NamedRange()
+    nr1.range = createGridRange(id: 50,
+                                startCol: 5 /* F (0 based) */,
+                                endCol: 6 /* F */,
+                                startRow: 7 /* 8 (0 based) */,
+                                endRow: 8 /* 8 */)
+    nr1.name = "ClearedSymbols"
+
+    let nr2 = GTLRSheets_NamedRange()
+    nr2.range = createGridRange(id: 42,
+                                startCol: 10 /* K */,
+                                endCol: 11 /* K */,
+                                startRow: 12 /* 13 */,
+                                endRow: 13 /* 13 */)
+    nr2.name = "MonthAndYear"
+
+    let nr3 = GTLRSheets_NamedRange()
+    nr3.range = createGridRange(id: 50,
+                                startCol: 35 /* AJ */,
+                                endCol: 40 /* AN */,
+                                startRow: 23 /* 24 */,
+                                endRow: 28 /* 28 */)
+    nr3.name = "TransactionCategories"
+
+    let mockSpreadsheet = MockSpreadsheet(sheets: [sheet1, sheet2])
+    mockSpreadsheet.namedRanges = [nr1, nr2, nr3]
+    let mockService = GTLRService.mockService(withFakedObject: mockSpreadsheet, fakedError: nil)
+    let query = GTLRSheetsQuery_SpreadsheetsGet.query(withSpreadsheetId: "abc")
+
+    let validator = GoogleSheetsValidator(sheetsService: mockService, sheetsQuery: query)
+    validator.validate(file: mockFile, for: mockUser)
+
+    let exp = XCTestExpectation()
+    exp.expectedFulfillmentCount = 2
+    validatorSink = validator.currentState.sink {
+      switch $0 {
+      case .dataMapRetrieved(let dataMap):
+        XCTAssertNotNil(dataMap["MonthAndYear"])
+        XCTAssertEqual(dataMap["MonthAndYear"]!, "Sheet2!K13:K13")
+        XCTAssertNotNil(dataMap["ClearedSymbols"])
+        XCTAssertEqual(dataMap["ClearedSymbols"]!, "Sheet1!F8:F8")
+        XCTAssertNotNil(dataMap["TransactionCategories"])
+        XCTAssertEqual(dataMap["TransactionCategories"]!, "Sheet1!AJ24:AN28")
+        exp.fulfill()
+
+      case .isLoading:
+        exp.fulfill()
+        
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/aspirebudgetingmobile/aspirebudgeting_ios/blob/fe2ab9e67c981f7885c3eaa47f96e04debe28a34/Aspire BudgetingTests/GoogleSheetsValidatorTests.swift#L148"></span>Lines should not have trailing whitespace.<br />
<code>trailing_whitespace</code> <code>GoogleSheetsValidatorTests.swift:148</code></td>
    </tr>
  </tbody>
</table>